doc/diffs.md

```diff
@@ -68,53 +166,184 @@ The "jit-diff list" command has this help message:
+4. `--core_root` -- specify the `CORE_ROOT` directory (the "test layout"). Used to specify to crossgen
+   where the platform assemblies are. Also, used as the directory where framework assemblies such as
+   System.Private.CoreLib.dll can be found for the purpose of using them to generate dasm.
+5. `--base` -- specify the directory in which a baseline crossgen can be found.
+6. `--diff` -- specify the directory in which a diff (experimental) crossgen can be found.
 
-Note: you create the `CORE_ROOT` directory "layout" by running the runtest script.
-On Windows, this can be created by running
+You create the `CORE_ROOT` directory "layout" by running the runtest script.
+On Windows, this can be created by running the following in the dotnet/coreclr repo root.
 ```
-    tests\runtest.cmd
+    c:\gh\coreclr> tests\runtest.cmd
 ```
 or
 ```
-    tests\runtest.cmd GenerateLayoutOnly
+    c:\gh\coreclr> tests\runtest.cmd GenerateLayoutOnly
 ```
-in the dotnet/coreclr repo. On non-Windows, consult the test instructions
+On non-Windows, consult the test instructions
 [here](https://github.com/dotnet/coreclr/blob/master/Documentation/building/unix-test-instructions.md).
 Note that you can pass `--testDir=NONE` to runtest.sh to get the
 same effect as passing `GenerateLayoutOnly` to runtest.cmd on Windows.
 
-## jit-analyze
+This command will generate the diffs into the specified output directory. It will automatically create a "tag"
+subdirectory (if the `--tag` switch isn't specified to override the default name), and within that tag
+directory will be "base" and "diff" directories, containing the diffs. The default tag directory looks like
+`dasmset_12`, with a unique number for every run of jit-diff.
 
-The jit-analyze tool understands the format of the `*.dasm` files in a diff and can extract
-this data to produce a summary and/or dump the info to a data file (in CSV or JSON format).
-The common usage of the tool is to extract interesting diffs, if any, from a diff run as
-part of the development progresses.
+jit-analyze will be run to compare the generated output.
 
-This tool can be invoked automatically after generating diffs using the `-a` option to `jit-diff diff`.
+You can also run a recursive textual comparison tool like windiff on Windows to visually compare
+the diffs, e.g.:
+```
+    c:\gh\coreclr> windiff c:\diffs\dasmset_12\base c:\diffs\dasmset_12\diff
+```
 
-Sample help command line:
+## Simplified jit-diff usage: defaults
+
+As seen above, specifying all required information can be quite verbose. jit-diff can automatically determine
+most of the arguments using computed defaults. The above diff can be accomplished using simply:
+```
+    c:\gh\coreclr> jit-diff diff --base_root e:\gh\coreclr2
+```
+
+The defaults are:
+* If jit-diff is invoked with the current directory within the dotnet/coreclr repo, then the root of
+  this repo serves to find the diff compiler and `CORE_ROOT` directory. (Note that we have no reasonable
```

Comment:
No, I was suggesting that `--base` and `--diff` would take arguments (a path) as today, which would do what they do today, but the argument would be optional. If you didn't pass a path, it would try to use a default. It wouldn't depend on `--diff_root` being set, for example, because you might want to set and leave that set in the config.json file.

So, you would have:
```
jit-diff diff
  -- illegal: no --base or --diff specified, so don't know what to diff
jit-diff diff --diff
  -- do a "diff" generation. Needs to use a default to find the diff JIT. It either uses the current
      directory or `--diff_root` in config.json to find the repo root, then uses default arch/build
      (optionally overridden by `--arch`, `--build` to find the JIT compiler.)
jit-diff diff --base
   -- do a "baseline" asm generation.
jit-diff diff --diff --base
   -- do both baseline and diff generation
jit-diff diff --base e:\gh\coreclr2\bin\Product\Windows_NT.x64.checked
   -- do just a baseline generation, using my given directory
jit-diff diff -diff --base e:\gh\coreclr2\bin\Product\Windows_NT.x64.checked
   -- do both diff and baseline generation, using my given directory for the baseline, and
      discovered defaults for the diff JIT.
```

Presumably only if both `--base` and `--diff` are specified could you invoke jit-analyze. Unless there are reasonable base/diff defaults that could be used. E.g., if you (a) generate baselines, then (b) sequentially generate a series of diff compiles -- could jit-analyze automatically (or non-automatically) find the baseline originally created in (a) for each run of (b)?